mslib/msui/remotesensing_dockwidget.py

```diff
@@ -316,10 +316,12 @@ def direction_coordinates(self, gc_lines):
         Returns: List of tuples of longitude/latitude coordinates
 
         """
-        lins = [(_line[0][mid], _line[0][mid + 1],
-                 _line[1][mid], _line[1][mid + 1]) for _line, mid in
-                zip(gc_lines, [len(_line[0]) // 2 for _line in gc_lines])]
-        lens = [np.hypot(_line[0][0] - _line[0][-1], _line[0][0] - _line[0][-1]) * 110. for _line in gc_lines]
+        lins = [(_line[0][mid], _line[0][mid + 1], _line[1][mid], _line[1][mid + 1])
```

Comment:
as bug fix okay, but that is not getting better readable. When there are tests written may be some of the methods can be  eextracted as functions to a remotesensing utils

Reply:
I agree.
However, I did also not see how to clean this up easily, or how to rewrite this using "numpy" terminology. The whole thing should be very slow and I am surprised it is still real-time capable.